File: source/java/nl/esciencecenter/xenon/examples/filesystems/DirectoryListing.java
.. code-tab:: java

    import nl.esciencecenter.xenon.filesystems.FileSystem;
    import nl.esciencecenter.xenon.filesystems.Path;
    import nl.esciencecenter.xenon.filesystems.PathAttributes;

    public class DirectoryListing {

        public static void main(String[] args) throws Exception {

            String adaptor = "file";
            FileSystem filesystem = FileSystem.create(adaptor);

            Path dir = new Path("/home/tutorial/xenon");
            boolean recursive = false;

            Iterable<PathAttributes> listing = filesystem.list(dir, recursive);

            for (PathAttributes elem : listing) {
                if (!elem.isHidden()) {
                    System.out.println(elem.getPath());
                }
            }

        }
    }

File: readthedocs/code-tabs/python/pyxenon_snippets/directory_listing.py
import xenon
from xenon import Path, FileSystem

def run_example():

    xenon.init(port=50051, disable_tls=True)

    filesystem = FileSystem.create(adaptor='file')
    path = Path("/home/travis/fixtures")

    listing = filesystem.list(path, recursive=False)

    for entry in listing:
        if not entry.path.is_hidden():
            print(entry.path)

    filesystem.close()


if __name__ == '__main__':
    run_example()
